File: HW_01/src/com/kodokushi/crud/Table.java
package com.kodokushi.crud;

import java.util.*;

public class Table {
    private final ArrayList<ArrayList<String>> table = new ArrayList<>();
    private final LinkedHashMap<String, Integer> columnNames = new LinkedHashMap<>();
    private int rowCounter = 0;

    public Table(ArrayList<String> columnNames) {
        for (int i = 0; i < columnNames.size(); i++) {
            this.columnNames.put(columnNames.get(i), i);
        }
    }

    public ArrayList<String> getColumnNames() {
        return new ArrayList<>(this.columnNames.keySet());
    }

    public int getRowCounter() {
        return rowCounter;
    }

    // ================================== CREATE ================================== //
    public void createRow(ArrayList<String> row) {
        rowCounter++;
        this.table.add(row);
    }

    // ================================== READ ================================== //
    public String readValue(int rowId, String columnName) {
        return this.table.get(rowId).get(columnNames.get(columnName));
    }
    public ArrayList<String> readRow(int rowId) {
        return this.table.get(rowId);
    }
    public ArrayList<ArrayList<String>> readTable() {
        return this.table;
    }

    // ================================== UPDATE ================================== //
    public void updateValue(int rowId, String columnName, String value) {
        this.table.get(rowId).set(columnNames.get(columnName), value);
    }
    public void updateRow(int rowId, ArrayList<String> row) {
        this.table.set(rowId, row);
    }

    // ================================== DELETE ================================== //
    public void deleteRow(int rowId) {
        this.table.remove(rowId);
    }
}

File: HW_05/table/table.go
package table

import "fmt"

type Table struct {
	name string
	table [][]string
	columnCounter int
}

func CreateTable(tableName string, columnNames []string) *Table {
	t := Table {
		name: tableName,
		columnCounter: len(columnNames),
	}
	t.table = append(t.table, columnNames)
	return &t
}

func (t *Table) CreateRow(rowValues []string) {
	if len(rowValues) == t.columnCounter {
		t.table = append(t.table, rowValues)
	} else {
		fmt.Println("Не удалось вставить строку: слишком много/мало аргументов")
	}
}

func (t *Table) ReadRow(rowId int) {
	if rowId < len(t.table) && rowId > 0 {
		for _, element := range t.table[rowId] {
			fmt.Print("\t", element)
		}
		fmt.Println()
	} else {
		fmt.Println("Не удалось считать строку: строки ", rowId, "не существует")
	}
}

func (t *Table) ReadTable() {
	for i := 0; i < len(t.table); i++ {
		for j := 0; j < t.columnCounter; j++ {
			fmt.Print("\t", t.table[i][j])
		}
		fmt.Println()
	}
}

func (t *Table) UpdateRow(rowId int, rowValues []string) {
	if len(rowValues) == t.columnCounter {
		if rowId < len(t.table) && rowId > 0 {
			t.table[rowId] = rowValues
		} else {
			fmt.Println("Не удалось обновить строку: строки", rowId, "не существует")
		}
	} else {
		fmt.Println("Не удалось обновить строку:  слишком много/мало аргументов")
	}

}

func (t *Table) DeleteRow(rowId int) {
	if rowId < len(t.table) && rowId > 0 {
		t.table = append(t.table[:rowId], t.table[rowId + 1:]...)
	} else {
		fmt.Println("Не удалось удалить строку: строки", rowId, "не существует")
	}
}